### ISSUE TITLE
`Allow directory as link targets` hangs IDE

### ISSUE THREAD

**Issue by vjpr** [vsch/idea-multimarkdown#600]
I put a simple line in a markdown file like:

```
See [packages/foo/docs](packages/foo/docs).
```

It hangs the IDE for > 10 minutes. I didn't see anything in the IDEA log.

I really would like to use this feature.

This is in a lerna monorepo project so there are many nested `node_modules` folders (but they are excluded).

Any idea what could be causing this hang?

Anyway to see debug logging info?

**Comment by vsch** [comment]
@vjpr, 

**Q:** What IDE and version are you using?

In addition to exceptions the IDE catches thread deadlocks and slow UI response. If you go to the logs directory using Help > Shot Log In Finder (or similar on other OS)

<img width="353" alt="image" src="https://user-images.githubusercontent.com/10299086/40724099-40462dd8-63ee-11e8-980d-a186f0568b7d.png">

<img width="820" alt="image" src="https://user-images.githubusercontent.com/10299086/40724485-2a69bae2-63ef-11e8-8a06-873ec9255483.png">

If you see a lot of thread dump directories it means there is something locking the UI.

I had this happen on 2018.2 EAP after a bit of use, Invalidating caches and restarting would help for a while but then it would happen again. Never 10 minutes. 20-30 seconds every time I tried to type anything, not just in Markdown. I switched to 2018.1 until it was fixed in the next release.

The node_modules should not be a problem. I have a couple of WebStorm projects with multiple top level modules each of which has its own node_modules directory and it is not causing issues.
